python/singa/utils.py

```diff
@@ -45,3 +46,395 @@ def update_progress(progress, info):
+            raise Exception('k must be smaller in [0,%d(length of size_list)]' % len(size_list))
+        for img in self.imgs:
+            if k ==0 or k== len(size_list):
+                small_sizes = size_list
+            else:
+                small_sizes = get_list_sample(size_list,k) 
+
+            for small_size in small_sizes:
+                new_img= do_resize(img,small_size)
+                new_imgs.append(new_img)
+        if update:
+            self.imgs=new_imgs
+            return self
+        else:
+            return new_imgs
+
+    def resize_for_test(self,(begin,end)):
+        size_list=[begin,begin+(end-begin)/2,end]
+        return self.resize_by_list(size_list,k=3)
+
+    def rotate_by_range(self,(begin,end),step=1,k=1,update=True):
+
+        '''
+        Args:
+            k: number of samples, must be smaller than or equare to the length of 
+               the range list, if k=0, then sample all
+            update: update imgs or not ( return new_imgs)
+        '''
+        size_list = range(begin,end+1,step)
+        return self.rotate_by_list(angle_list,k,update)
+        
+    def rotate_by_list(self,angle_list,k=1,update=True):
+        '''
+        Args:
+            k: number of samples, must be smaller than or equare to the length of 
+               size_list, if k=0, then sample all
+            update: update imgs or not ( return new_imgs)
+        '''
+        new_imgs=[]
+        if k<0 or k > len(angle_list):
+            raise Exception('k must be smaller in [0,%d(length of angle_list)]' % len(angle_list))
+
+        for img in self.imgs:
+            if k ==0 or k== len(angle_list):
+                angles = angle_list
+            else:
+                angles = get_list_sample(angle_list,k) 
+
+            for angle in angles:
+                new_img= img.rotate(angle)
+                new_imgs.append(new_img)
+        if update:
+            self.imgs=new_imgs
+            return self
+        else:
+            return new_imgs
+
+
+    def crop_5(self,crop_size,k=1,update=True):
+        '''
+        Args:
+            k: number of samples, must be in [0,5], if k=0, then sample all
+            update: update imgs or not ( return new_imgs)
+        '''
+        new_imgs=[]
+        positions=["left_top","left_bottom","right_top","right_bottom","center"]
+        if k > 5 or k <0:
+            raise Exception('k must be in [0,5]')
+        for img in self.imgs:
+
+            if k>0 and k<5:
+                positions = get_list_sample(positions,k) 
+
+            for position in positions:
+                new_img=do_crop(img,crop_size,position)
+                new_imgs.append(new_img)
+
+        if update:
+            self.imgs=new_imgs
+            return self
+        else:
+            return new_imgs
+
+    def crop_and_scale(self,crop_size,k=1,update=True):
+        '''
+        Args:
+            k: number of samples, must be in [0,3], if k=0, then sample all
```

Comment:
add comments:
crop and scale the image into the given size.
crop position could be either (left, center, right) or (up, center, low).